presto-parquet/src/test/java/io/prestosql/parquet/predicate/TestPredicateUtils.java

```diff
@@ -0,0 +1,107 @@
+import static org.apache.parquet.column.Encoding.RLE_DICTIONARY;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestPredicateUtils
+{
+    @Test
+    @SuppressWarnings("deprecation")
+    public void testDictionaryEncodingCasesV1()
+    {
+        Set<Encoding> required = ImmutableSet.of(BIT_PACKED);
+        Set<Encoding> optional = ImmutableSet.of(BIT_PACKED, RLE);
+        Set<Encoding> repeated = ImmutableSet.of(RLE);
+
+        Set<Encoding> notDictionary = ImmutableSet.of(PLAIN);
+        Set<Encoding> mixedDictionary = ImmutableSet.of(PLAIN_DICTIONARY, PLAIN);
+        Set<Encoding> dictionary = ImmutableSet.of(PLAIN_DICTIONARY);
+
+        assertFalse(isOnlyDictionaryEncodingPages(newV1ColumnMetaData(union(required, notDictionary))),
+                "required notDictionary");
+        assertFalse(isOnlyDictionaryEncodingPages(newV1ColumnMetaData(union(optional, notDictionary))),
+                "optional notDictionary");
+        assertFalse(isOnlyDictionaryEncodingPages(newV1ColumnMetaData(union(repeated, notDictionary))),
+                "repeated notDictionary");
+        assertFalse(isOnlyDictionaryEncodingPages(newV1ColumnMetaData(union(required, mixedDictionary))),
+                "required mixedDictionary");
+        assertFalse(isOnlyDictionaryEncodingPages(newV1ColumnMetaData(union(optional, mixedDictionary))),
+                "optional mixedDictionary");
+        assertFalse(isOnlyDictionaryEncodingPages(newV1ColumnMetaData(union(repeated, mixedDictionary))),
+                "repeated mixedDictionary");
+        assertTrue(isOnlyDictionaryEncodingPages(newV1ColumnMetaData(union(required, dictionary))),
+                "required dictionary");
+        assertTrue(isOnlyDictionaryEncodingPages(newV1ColumnMetaData(union(optional, dictionary))),
+                "optional dictionary");
+        assertTrue(isOnlyDictionaryEncodingPages(newV1ColumnMetaData(union(repeated, dictionary))),
+                "repeated dictionary");
+    }
+
+    @Test
+    @SuppressWarnings("deprecation")
+    public void testDictionaryEncodingCasesV2()
+    {
+        assertTrue(isOnlyDictionaryEncodingPages(newV2ColumnMetaData(RLE_DICTIONARY)));
+        assertTrue(isOnlyDictionaryEncodingPages(newV2ColumnMetaData(PLAIN_DICTIONARY)));
+
+        assertFalse(isOnlyDictionaryEncodingPages(newV2ColumnMetaData(PLAIN)));
+
+        // Simulate fallback to plain encoding e.g. too many unique entries
+        assertFalse(isOnlyDictionaryEncodingPages(newV2ColumnMetaData(RLE_DICTIONARY, PLAIN)));
+    }
+
+    private ColumnChunkMetaData newV2ColumnMetaData(Encoding... dataEncodings)
+    {
+        EncodingStats encodingStats = new EncodingStats.Builder()
+                .withV2Pages()
+                .addDictEncoding(PLAIN)
+                .addDataEncodings(ImmutableSet.copyOf(dataEncodings)).build();
+
+        return ColumnChunkMetaData.get(ColumnPath.fromDotString("column"),
```

Comment:
static import ColumnPath.fromDotString, PrimitiveType.PrimitiveTypeName.BINARY, CompressionCodecName.UNCOMPRESSED, try make it a single line